test/libsolidity/SyntaxTest.cpp

```diff
@@ -207,6 +157,71 @@ void SyntaxTest::printSource(ostream& _stream, string const& _linePrefix, bool _
 		}
 }
 
+void SyntaxTest::setupCompiler()
+{
+	string const versionPragma = "pragma solidity >=0.0;\n";
+	compiler().reset();
+	auto sourcesWithPragma = m_sources;
+	for (auto& source: sourcesWithPragma)
+		source.second = versionPragma + source.second;
+	compiler().setSources(sourcesWithPragma);
+	compiler().setEVMVersion(m_evmVersion);
+	compiler().setParserErrorRecovery(m_parserErrorRecovery);
+	compiler().setOptimiserSettings(
+		m_optimiseYul ?
+		OptimiserSettings::full() :
+		OptimiserSettings::minimal()
+	);
+}
+
+void SyntaxTest::parseAndAnalyze()
+{
+	if (compiler().parse())
```

Comment:
why not `if (compiler().parsen) && compiler.analyze())`

Reply:
I just copied that while splitting the content into new functions

Reply:
Combining them seems to make sense.